heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -283,33 +275,55 @@ export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
       },
       jsonPathMetadata: {
         // string
-        '$.dependencyExtractor': nodeResolveMetadata,
-        '$.filter': nodeResolveMetadata,
-        '$.globalSetup': nodeResolveMetadata,
-        '$.globalTeardown': nodeResolveMetadata,
-        '$.moduleLoader': nodeResolveMetadata,
-        '$.prettierPath': nodeResolveMetadata,
-        '$.resolver': nodeResolveMetadata,
-        '$.runner': nodeResolveMetadata,
-        '$.snapshotResolver': nodeResolveMetadata,
+        '$.cacheDirectory': tokenResolveMetadata,
+        '$.coverageDirectory': tokenResolveMetadata,
+        '$.dependencyExtractor': jestResolveMetadata,
+        '$.filter': jestResolveMetadata,
+        '$.globalSetup': jestResolveMetadata,
+        '$.globalTeardown': jestResolveMetadata,
+        '$.moduleLoader': jestResolveMetadata,
+        '$.prettierPath': jestResolveMetadata,
+        '$.resolver': jestResolveMetadata,
+        '$.runner': JestPlugin._getJsonPathMetadata({
+          buildFolder,
+          resolveAsModule: true,
+          modulePrefix: 'jest-runner-',
+          ignoreMissingModule: true
+        }),
+        '$.snapshotResolver': jestResolveMetadata,
         // This is a name like "jsdom" that gets mapped into a package name like "jest-environment-jsdom"
-        // '$.testEnvironment': string
-        '$.testResultsProcessor': nodeResolveMetadata,
-        '$.testRunner': nodeResolveMetadata,
-        '$.testSequencer': nodeResolveMetadata,
+        '$.testEnvironment': JestPlugin._getJsonPathMetadata({
```

Comment:
For these it would be useful to mention [resolveWithPrefix()](https://github.com/facebook/jest/blob/d6fb0d8fb0d43a17f90c7a5a6590257df2f2f6f5/packages/jest-resolve/src/utils.ts#L36) and then cite each Jest rule like [resolveTestEnvironment](https://github.com/facebook/jest/blob/d6fb0d8fb0d43a17f90c7a5a6590257df2f2f6f5/packages/jest-resolve/src/utils.ts#L110).  

This would be very valuable if a future version of Jest changes the logic, and someone needs to figure out what changed since Heft's implementation was put together.